src/utils/classes/FormValidator.ts

```diff
@@ -0,0 +1,149 @@
+    this._inputSelector = inputSelector;
+    this._inputHelperTextSelector = inputHelperTextSelector;
+    this._isShowHelperTextSelector = isShowHelperTextSelector;
+    this._isDisableBtnSubmitSelector = config.isDisableBtnSubmitSelector;
+    this._errorContainer = null;
+    this._isValidFieldWithCustomRules = null;
+  }
+
+  private _closeErrorMessage() {
+    if (this._errorContainer) {
+      this._errorContainer.textContent = '';
+      this._errorContainer.classList.remove(this._isShowHelperTextSelector);
+    }
+  }
+
+  private _showErrorMessage(validationMessage: string) {
+    if (this._errorContainer) {
+      this._errorContainer.textContent = validationMessage;
+      this._errorContainer.classList.add(this._isShowHelperTextSelector);
+    }
+  }
+
+  private _getInputFromForm = () => {
+    const form = document.querySelector(`.${this._formSelector}`) as HTMLFormElement;
+    return form.querySelectorAll(`.${this._inputSelector}`);
+  };
+
+  private _checkExpressionByRegExp = (regexp: RegExp, value: string) => {
+    return !regexp.test(value);
+  };
+
+  private _checkFieldByCustomRules = (element: EventTarget | null) => {
+    const input = element as HTMLFormElement;
+    if (
+      input &&
+      (input.name === USER_NAME_FIELD || input.name === LAST_NAME_USER_FIELD)
+    ) {
+      const isValidValue = this._checkExpressionByRegExp(
+        REGEXP_FOR_NAME_AND_LASTNAME,
+        input.value
+      );
+      isValidValue && this._showErrorMessage(CUSTOM_ERROR_FOR_NAME_AND_LASTNAME);
+      this._isValidFieldWithCustomRules = {
+        ...this._isValidFieldWithCustomRules,
+        [input.name]: isValidValue,
+      };
+    }
+    if (input && input.name === PHONE_USER_FIELD) {
+      const isValidValue = this._checkExpressionByRegExp(REGEXP_FOR_PHONE, input.value);
+      isValidValue && this._showErrorMessage(CUSTOM_ERROR_FOR_PHONE_FILED);
+      this._isValidFieldWithCustomRules = {
+        ...this._isValidFieldWithCustomRules,
+        [input.name]: isValidValue,
+      };
+    }
+  };
+
+  private _checkValidFieldsWithCustomRules = () => {
+    if (this._isValidFieldWithCustomRules) {
+      return Object.values(this._isValidFieldWithCustomRules).find(
+        (element) => element === true
+      );
+    }
+  };
+
+  public handleFieldValidation(evt: Event) {
+    const element = evt.target;
+
+    this._errorContainer = (<Element>(
+      element
+    ))?.parentElement?.parentElement?.querySelector(`.${this._inputHelperTextSelector}`);
+
+    !(<HTMLFormElement>element).validity.valid
+      ? this._showErrorMessage((<HTMLFormElement>element).validationMessage)
+      : this._closeErrorMessage();
+
+    this._checkFieldByCustomRules(element);
+
+    this._checkValidFieldsWithCustomRules() && this.disableBtn();
+  }
+
+  public clearError() {
+    this._closeErrorMessage();
+  }
+
+  public checkStateForm() {
+    const form = document.querySelector(`.${this._formSelector}`) as HTMLFormElement;
```

Comment:
Лучше стараться избегать явного приведения типов, если это возможно
```suggestion
    const form = document.querySelector<HTMLFormElement>(`.${this._formSelector}`);
```